src/component.js

```diff
@@ -1,146 +1,87 @@
 // Base component that handles comm messages and renders components to notebook cell
-import react from 'react';
-import reactDom from 'react-dom';
+import React from 'react';
+import autobind from 'autobind-decorator';
 
-module.exports = function Component( options ) {
-  return function (comm, props, cell) {
-    this.cell = cell;
-    this.comm = comm;
+const { bool, object } = React.PropTypes;
 
-    /**
-     * handleMsg 
-     * Handle all messages over this comm
-     */
-    this.handleMsg = msg => {
-      const data = msg.content.data;
-      switch (data.method) {
-        case "update":
-          if ( options.on_update ) {
-            return options.on_update( props.content.data.module, data.props, msg.content.comm_id);
-          }
-          // else re-render
-          this.render( msg, data.props );
-          break;
-        case "display":
-          // save comm id and cell id to notebook.metadata
-          if ( options.save ) {
-            this._save( msg, () => {
-              this.render( msg, data.props );
-            } );
-          } else {
-            this.render( msg, data.props );
-          }
-          break;
-      }
-    };
-
-    /**
-     * _save
-     * save cell index to notebook metadata as a string
-     */
-    this._save = function( msg, done ) {
-      const cell = this._getMsgCell( msg );
-      const md = Jupyter.notebook.metadata;
-      if ( cell ) {
-        if ( !md.react_comms ) {
-          md.react_comms = {}
-        }
-        md.react_comms[ comm.comm_id ] = this._getCellIndex( cell.cell_id ) + '';
-      }
-      done();
-    };
+const propTypes = {
+  comm:       object.isRequired,
+  comm_msg:   object.isRequired,
+  components: object.isRequired,
+  save:       bool
+};
 
-  
-    /**
-     * render
-     * appends the components to the dom
-     *
-     */
-    this.render = function( msg, _newProps ) {
-      const newProps = _newProps || props.content.data.props || props.content.data;
-      newProps.cell = this.cell || this._getMsgCell( msg );
-      newProps.comm = comm;
+@autobind
+class Component extends React.Component {
 
-      let display;
-      const domId = props.content.data.domId;
+  constructor( props ) {
+    super( props );
+    this.state = {
+      renderProps: null,
+      components: props.components,
+      comm: props.comm,
+      comm_msg: props.comm_msg,
+      save: props.save
+    };
```

Comment:
@chelm Do all of these need to be on state? It seems like just `renderProps` would need to be there.